translations/extensions/io_xls.py

```diff
@@ -0,0 +1,209 @@
+        file_name = shutil.make_archive(Constants.DIST_PATH + self.out_name + '_' + str(timestamp),
+                                        'zip',
+                                        Constants.DEFAULT_TRANSL_XLS_PATH)
+        print(f'Package generated in "{file_name}"')
+
+    def get_locale_out_target(self, locale):
+        if locale in self.export_mapping:
+            return self.export_mapping[locale]
+        return locale
+
+    def write_xls(self, locale, translations, postfix=None, context_col_value=None):
+        workbook = Workbook()
+        sheet = workbook.active
+
+        sheet['A1'] = 'en_US'
+        sheet['B1'] = locale
+        sheet['C1'] = 'CONTEXT'
+        sheet['D1'] = 'CONTEXT_DESCRIPTION'
+        row = 2
+
+        for translation in translations:
+            sheet[f'A{row}'] = translation
+            sheet[f'C{row}'] = context_col_value if context_col_value else ''
+            row += 1
+
+        workbook.save(Constants.DEFAULT_TRANSL_XLS_PATH + locale + ('-' + postfix if postfix else '') + '.xls')
+
+
+class XlsImporter(TranslationResponseProcessor):
+    whoami = __qualname__
+
+    def __init__(self, config, options):
+        self.default_locale = ConfigUtilities.get_value(config, ('locales', 'default'))
+        self.supported_locales = ConfigUtilities.get_value(config, ('locales', 'supported'))
+        self.translations_pkg = ConfigUtilities.get_value(config, ('io', 'in', 'package'))
+        self.import_mapping = ConfigUtilities.get_value(config, ('io', 'in', 'mapping'))
+        self.expected_locales = self.determine_expected_locales()
+        self.options = options
+
+    def determine_expected_locales(self):
+        expected_locales = self.supported_locales.copy()
+        for inbound_locale, mapped_locales in self.import_mapping.items():
+            expected_locales = [locale for locale in expected_locales if locale not in mapped_locales]
+            if inbound_locale not in expected_locales:
+                expected_locales.append(inbound_locale)
+        return expected_locales
+
+    def determine_inbound_locale(self, supported_locale):
+        for inbound_locale, mapped_locales in self.import_mapping.items():
+            for mapped_locale in mapped_locales:
+                if mapped_locale == supported_locale:
+                    return inbound_locale
+        return supported_locale
+
+    def process_response(self, manifest):
+        import_manifest = {}
+        new_messages = {}
+        missing = manifest.data.get('missing')
+        added = manifest.data.get('added')
+        translations = XlsTranslationsProcessor.get_inbound_translations(self.translations_pkg,
+                                                                         self.default_locale,
+                                                                         self.expected_locales)
+        if self.options.dump:
+            Utilities.write_to_json_file('translations-raw', translations)
```

Comment:
I think it's probably more readable to specify the full qualified filename here ex: `translation-raw.json` instead of appending the json extension in the method?